From a1aa5f3cbd567779ff853cbf8c84bd04db0fdbc8 Mon Sep 17 00:00:00 2001
From: Oliver Beard <olib141@outlook.com>
Date: Fri, 16 Jan 2026 22:47:32 +0000
Subject: [PATCH 5/5] FROMLIST: lockscreen,sddm-theme: Adapt to WallpaperFader
 changes

WallpaperFader has had greeter functionality split into GreeterFader, which we should use instead.

Link: https://invent.kde.org/plasma/plasma-desktop/-/merge_requests/3460
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 .../contents/lockscreen/LockScreenUi.qml          | 13 +++++--------
 sddm-theme/Main.qml                               | 15 +++++++--------
 2 files changed, 12 insertions(+), 16 deletions(-)

diff --git a/desktoppackage/contents/lockscreen/LockScreenUi.qml b/desktoppackage/contents/lockscreen/LockScreenUi.qml
index 4640fa1932..9487aeb5f5 100644
--- a/desktoppackage/contents/lockscreen/LockScreenUi.qml
+++ b/desktoppackage/contents/lockscreen/LockScreenUi.qml
@@ -198,8 +198,12 @@ Item {
 
         WallpaperFader {
             anchors.fill: parent
-            state: lockScreenRoot.uiVisible ? "on" : "off"
+            factor: lockScreenRoot.uiVisible ? 1 : 0
             source: wallpaper
+        }
+
+        GreeterFader {
+            state: lockScreenRoot.uiVisible ? "on" : "off"
             mainStack: mainStack
             footer: footer
             clock: clock
@@ -216,13 +220,6 @@ Item {
             samples: 15
             spread: 0.2
             color : Qt.rgba(0, 0, 0, 0.7)
-            opacity: lockScreenRoot.uiVisible ? 0 : 1
-            Behavior on opacity {
-                OpacityAnimator {
-                    duration: Kirigami.Units.veryLongDuration * 2
-                    easing.type: Easing.InOutQuad
-                }
-            }
         }
 
         Clock {
diff --git a/sddm-theme/Main.qml b/sddm-theme/Main.qml
index 210ac01667..9a8a42a890 100644
--- a/sddm-theme/Main.qml
+++ b/sddm-theme/Main.qml
@@ -102,14 +102,20 @@ Item {
                 }
             }
         }
+
         WallpaperFader {
             visible: config.type === "image"
             anchors.fill: parent
-            state: loginScreenRoot.uiVisible ? "on" : "off"
+            factor: loginScreenRoot.uiVisible ? 1 : 0
             source: wallpaper
+        }
+
+        GreeterFader {
+            state: loginScreenRoot.uiVisible ? "on" : "off"
             mainStack: mainStack
             footer: footer
             clock: clock
+            alwaysShowClock: true
         }
 
         DropShadow {
@@ -122,13 +128,6 @@ Item {
             samples: 15
             spread: 0.2
             color : Qt.rgba(0, 0, 0, 0.7)
-            opacity: loginScreenRoot.uiVisible ? 0 : 1
-            Behavior on opacity {
-                OpacityAnimator {
-                    duration: Kirigami.Units.veryLongDuration * 2
-                    easing.type: Easing.InOutQuad
-                }
-            }
         }
 
         Clock {
-- 
2.52.0

